kernel_regress test failure when cross building for 32 bit

hello

apologies if I'm missing some blatant point here.

I'm on linux, AMD Athlon (specs enclosed below)

When cross building 0.3.7 for 32 bit on a 64 bit machine I'm getting
the following test failure:

    ----8<----
    TEST 23/24 kernel_regress:skx_avx  ** On entry to DGESVD parameter number  9 had an illegal value
    [FAIL]
      ERR: test_kernel_regress.c:49  expected 0.000e+00, got 5.747e+02 (diff -5.747e+02, tol 1.000e-10)
    TEST 24/24 fork:safety [OK]
    RESULTS: 24 tests (23 ok, 1 failed, 0 skipped) ran in 598 ms
    make[2]: *** [Makefile:37: run_test] Error 1
    make[2]: Leaving directory '/home/balducci/tmp/install-us-d/openblas-0.3.6.d/OpenBLAS-0.3.7/utest'
    ---->8----

64 bit build completes with no problems

I have been maintaining openblas since quite a bit (since 0.2.8) for
both 32 and 64 bit and never had problems

I just noticed that the failing test has been added in 0.3.7 (wasn't
there in 0.3.6)...

I'll be happy to send any other information and/or to perform any test
which might help

many thanks in advance for any hint/help
ciao
gabriele

My specs follow:
---------------
openblas:49> uname -sr
Linux 5.2.7

openblas:47> gcc --version
gcc (GCC) 9.1.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Lapack version: 3.8.0

32 bit make command:
make TARGET=BULLDOZER BINARY=32 'CC=gcc -m32' 'CXX=g++ -m32'

64 bit make command:
make TARGET=BULLDOZER BINARY=64 'CC=gcc -m64' 'CXX=g++ -m64'

openblas:50> /lib/libc.so.6 
GNU C Library (GNU libc) stable release version 2.29.
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 8.2.0.
libc ABIs: UNIQUE IFUNC ABSOLUTE
For bug reporting instructions, please see:
<http://www.gnu.org/software/libc/bugs.html>.

openblas:58> cat /proc/cpuinfo
processor       : 0
vendor_id       : AuthenticAMD
cpu family      : 21
model           : 16
model name      : AMD Athlon(tm) X4 740 Quad Core Processor
stepping        : 1
microcode       : 0x6001116
cpu MHz         : 2449.157
cache size      : 2048 KB
physical id     : 0
siblings        : 4
core id         : 0
cpu cores       : 2
apicid          : 16
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 popcnt aes xsave avx f16c lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs xop skinit wdt lwp fma4 tce nodeid_msr tbm topoext perfctr_core perfctr_nb cpb hw_pstate ssbd vmmcall bmi1 arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold
bugs            : fxsave_leak sysret_ss_attrs null_seg spectre_v1 spectre_v2 spec_store_bypass
bogomips        : 6400.51
TLB size        : 1536 4K pages
clflush size    : 64
cache_alignment : 64
address sizes   : 48 bits physical, 48 bits virtual
power management: ts ttp tm 100mhzsteps hwpstate cpb eff_freq_ro

--- a/kernel/x86/lsame.S
+++ b/kernel/x86/lsame.S
@@ -56,13 +56,13 @@
 #ifndef HAVE_CMOV
 	movl	%eax, %ecx
 	subl	$32,  %ecx
-	jle	.L1
+	jl	.L1
 	movl	%ecx, %eax
 .L1:
 
 	movl	%edx, %ecx
 	subl	$32,  %ecx
-	jle	.L2
+	jl	.L2
 	movl	%ecx, %edx
 .L2:
 	subl	%eax, %edx

